### PR TITLE
[KubernetesConfiguration] Releasing dotnet SDK for KubernetesConfiguration.Extensions api version 2021-09-01

### PR DESCRIPTION
**Pull request by willdavsmith**
# Description
* Added dotnet SDK for KubernetesConfiguration Extensions version 2021-09-01
* Moved Extensions SDK to new package
* Moved SourceControlConfiguration to new package

[Swagger spec PR](https://github.com/Azure/azure-rest-api-specs/commit/fa0a95854a551be7fdb04367e2e7b6500ab2e341)

Note: some of the generated code was edited.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [x] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by Yao725 — COMMENTED** [review]
Hi @willdavsmith , it seems that you are trying to release two new packages for KubernetesConfiguration. However, we are not going to onboard new track 1 SDK package. Track 2 SDK is suggested and please talk to my manager @ArthurMa1978 for the track 2 onboard details.

File: sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -4,8 +4,8 @@
 using System.Reflection;
 using System.Resources;
 
-[assembly: AssemblyTitle("Microsoft Azure KubernetesConfiguration Management Library")]
-[assembly: AssemblyDescription("Provides Microsoft Azure KubernetesConfiguration management operations including the ability to create, update and delete runbooks and schedules.")]
+[assembly: AssemblyTitle("Microsoft Azure KubernetesConfiguration SourceControlConfigurations Management Library")]
+[assembly: AssemblyDescription("Provides Microsoft Azure KubernetesConfiguration SourceControlConfigurations management operations including the ability to create, update and delete configurations.")]
 
 [assembly: AssemblyVersion("1.0.0.0")]
```
Comment (Yao725) [inline sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs:10]: I want to double confirm with you that it is intentional change to split the namespace `Microsoft.Azure.Management.KubernetesConfiguration` into `Microsoft.Azure.Management.KubernetesConfiguration.SourceControlConfigurations` and `Microsoft.Azure.Management.KubernetesConfiguration.Extensions`. If so, you need to bump the major version here.

Comment (Yao725) [inline sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs:10]: I want to double confirm with you that it is intentional change to split the namespace `Microsoft.Azure.Management.KubernetesConfiguration` into `Microsoft.Azure.Management.KubernetesConfiguration.SourceControlConfigurations` and `Microsoft.Azure.Management.KubernetesConfiguration.Extensions`. If so, you need to bump the major version here.

File: sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -4,8 +4,8 @@
 using System.Reflection;
 using System.Resources;
 
-[assembly: AssemblyTitle("Microsoft Azure KubernetesConfiguration Management Library")]
-[assembly: AssemblyDescription("Provides Microsoft Azure KubernetesConfiguration management operations including the ability to create, update and delete runbooks and schedules.")]
+[assembly: AssemblyTitle("Microsoft Azure KubernetesConfiguration SourceControlConfigurations Management Library")]
+[assembly: AssemblyDescription("Provides Microsoft Azure KubernetesConfiguration SourceControlConfigurations management operations including the ability to create, update and delete configurations.")]
```
Comment (Yao725) [inline sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs:8]: Suggested changes as this package also contains KubernetesConfiguration Extension.
```suggestion
[assembly: AssemblyTitle("Microsoft Azure KubernetesConfiguration Management Library")]
[assembly: AssemblyDescription("Provides Microsoft Azure KubernetesConfiguration management operations including the ability to create, update and delete configurations and extensions.")]
```

Comment (willdavsmith) [reply sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs]: Updated in latest commit, PTAL

File: sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (Yao725) [reply sdk/kubernetesconfiguration/Microsoft.Azure.Management.KubernetesConfiguration/src/Properties/AssemblyInfo.cs:10]: Please bump the major version like this:
```
[assembly: AssemblyVersion("2.0.0.0")]
[assembly: AssemblyFileVersion("2.0.0.0")]
```

**Review by Yao725 — APPROVED** [review]
LGTM

**Pull request merged**